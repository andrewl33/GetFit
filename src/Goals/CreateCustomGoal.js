import React, { Component } from 'react';
import { StyleSheet, Text, TextInput, View, Button, Alert } from 'react-native';

/**
 * AsyncStorage is global,
 * so we're going to store Goal storage like
 * AsyncStorage: {
 *  Goals: {
 *      <GoalUniqueIdGoesHere>: {
 *        id: <values>,
 *        title: string,
 *        createDate:  <dateString>,
 *        endDate: <dateString>,
 *        isActive: boolean,
 *        unitType: steps | <someSortOfUnitToMeasure>,
 *        startAmount: number,
 *        progress: number,
 *        endAmount: number
 *     },
 *     activeGoals: ["<goalID>"]
 *  }
 * }
 */

/**
 * addNewGoal
 *
 * uses static setItem(key: string, value: string, [callback]: ?(error: ?Error) => void)
 *
 */

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'center',
        alignSelf: 'stretch',
        marginTop: 20,
    },
    header: {
        fontSize: 35,
    },
    button: {
        margin: 30,
    },
    row: {
        flex: 1,
        flexDirection: "row"
    },
    inputWrap: {
        flex: 1,
        borderColor: "#cccccc",
        borderBottomWidth: 1,
    }
});

export class UserTextInput extends Component {
    render() {
        return (
            <View style={styles.row}>
                <View style={styles.inputWrap}>
                    <Text>{ this.props.inputLabel }:</Text> 
                    <TextInput
                        placeholder="Type here"
                        onChangeText={(text) => this.setState({text})}
                    />
                </View>
            </View>
        );
    }
}

export default class CreateCustomGoal extends Component {

    AlertText = () => {
        Alert.alert('Custom Goal Added! ');
    };

    render() {
        return (
            <View style={styles.container}>
                <Text style={styles.header}>Create A Custom Goal</Text>
                <UserTextInput inputLabel='title' />
                <UserTextInput inputLabel='start date' />
                <UserTextInput inputLabel='end date' />
                <UserTextInput inputLabel='amount' />
                <UserTextInput inputLabel='unit' />
                <View style={styles.button}>
                    <Button onPress={() => this.AlertText()} title="Add Custom Goal">
                        Add Custom Goal
                    </Button>
                </View>
            </View>
        );
    }
}
