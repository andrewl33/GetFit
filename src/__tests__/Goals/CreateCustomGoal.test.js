import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import CreateCustomGoal from '../../Goals/CreateCustomGoal';

describe('CreateCustomGoal', () => {
  it('renders', () => {
    const tree = renderer.create(<CreateCustomGoal />);
    expect(tree).toMatchSnapshot();
  });
});
