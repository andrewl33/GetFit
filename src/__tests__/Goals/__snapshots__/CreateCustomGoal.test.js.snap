// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateCustomGoal renders 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "flex": 1,
      "justifyContent": "center",
      "marginTop": 20,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 35,
      }
    }
  >
    Create A Custom Goal
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="title"
    style={
      Object {
        "borderColor": "gray",
        "borderWidth": 1,
        "height": 40,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="start date"
    style={
      Object {
        "borderColor": "gray",
        "borderWidth": 1,
        "height": 40,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="end date"
    style={
      Object {
        "borderColor": "gray",
        "borderWidth": 1,
        "height": 40,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="amount"
    style={
      Object {
        "borderColor": "gray",
        "borderWidth": 1,
        "height": 40,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="unit"
    style={
      Object {
        "borderColor": "gray",
        "borderWidth": 1,
        "height": 40,
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    style={
      Object {
        "margin": 30,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Add Custom Goal
        </Text>
      </View>
    </View>
  </View>
</View>
`;
